libraries/AP_Airspeed/AP_Airspeed_SDP3X.h

```diff
@@ -0,0 +1,70 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+/*
+  backend driver for airspeed from I2C
+ */
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Param/AP_Param.h>
+#include <AP_HAL/utility/OwnPtr.h>
+#include <AP_HAL/I2CDevice.h>
+#include <utility>
+
+#include "AP_Airspeed_Backend.h"
+#include "AP_Airspeed.h"
+
+/*
+ * Thanks to PX4 for registers definitions
+ */
+
+class AP_Airspeed_SDP3X : public AP_Airspeed_Backend
+{
+public:
+    AP_Airspeed_SDP3X(AP_Airspeed &frontend, uint8_t _instance);
+    ~AP_Airspeed_SDP3X(void) {}
+
+    // probe and initialise the sensor
+    bool init() override;
+
+    // return the current differential_pressure in Pascal
+    bool get_differential_pressure(float &pressure) override;
+
+    // return the current temperature in degrees C, if available
+    bool get_temperature(float &temperature) override;
+
+private:
+    void _collect();
+    void _timer();
+    bool _send_command(uint16_t cmd);
+    float _get_pressure(int16_t dp_raw) const;
+    float _get_temperature(int16_t dT_raw) const;
+    bool _crc(const uint8_t data[], unsigned size, uint8_t checksum);
+    float _correct_pressure(float press);
+
+    float _temp;
+    float _press;
```

Comment:
This also appears to be just a local to one function?

Reply:
needed for zero counts